docs-ref-services/key-vault.md

```diff
@@ -142,18 +111,140 @@ operation = client.vaults.create_or_update(
+
+```python
+# Authentication with a service principal
+from azure.common.credentials import ServicePrincipalCredentials
+
+# Replace the values with your specific IDs
+client_id = "your_client_id"
+tenant_id = "your_tenant_id"
+app_secret = "your_app_secret"
+
+credentials = ServicePrincipalCredentials(client_id, app_secret, tenant_id)
+```
+
+With the `credentials` variable initialized, you can then use the following code to create a secret named `ExampleSecret` in an Azure Key Vault named `msdocs-vault01` (as created using the code in the [management library](#management-library) section.
+
+```python
+# You can place this import statement earlier in your code.
+from azure.keyvault.secrets import SecretClient
+
+# The URL of your Key Vault obtained from the Azure portal, which ends in vault.azure.net.
+# This example uses https://msdocs-vault01.vault.azure.net/
+key_vault_uri = "https://msdocs-vault01.vault.azure.net/"
+
+# Obtain the client object
+secret_client = SecretClient(key_vault_uri, credentials)
+
+# Create a secret. In production code deployed to a public server, such values
+# would not appear in source code.
+secret_name = "ExampleSecret"
+secret_client.set_secret(secret_name, "value5502")
+```
+
+The `set_secret` method updates the value if the secret already exists.
+
+Use the same `SecretClient` object to retrieve the secret value:
+
+```python
+secret = secret_client.get_secret(secret_name)
+```
+
+To delete the secret:
+
+```python
+client.begin_delete_secret(secret_name)
+```
+
+> [!div class="nextstepaction"]
+> [Explore the Client APIs](/python/api/overview/azure/keyvault/client)
+
+## Using Managed Identities locally
+
+Managed Identities are a feature of Azure hosting environments and are not inherently available on local developer workstations. As a result, attempting to run code that uses `DefaultAzureCredential` may produce the following error:
+
+<pre>
+No credential in this chain provided a token.
+Attempted credentials:
+        EnvironmentCredential: Incomplete environment configuration
+        ImdsCredential: IMDS endpoint unavailable
+        SharedTokenCacheCredential: The shared cache contains no signed-in accounts.
+        To authenticate with SharedTokenCacheCredential, login through developer tooling
+        supporting Azure single sign on.
+</pre>
+
+There are two options to run the code locally:
+
+- Use a developer tool, like Visual Studio, that supports a single sign-on experience for Azure and therefore provides a SharedTokenCacheCredential.
+
+- Configure a local service principal and create environment variables to provide the EnvironmentCredential:
+
+    1. Install the [Azure CLI](/cli/azure/install-azure-cli?view=azure-cli-latest) if you don't have it installed already.
+
+    1. In a terminal or command prompt, create a service principle using the Azure CLI [az ad sp create-for-rbac](/cli/azure/ad/sp?view=azure-cli-latest#az-ad-sp-create-for-rbac) command:
```

Comment:
```suggestion
    1. In a terminal or command prompt, create a service principal using the Azure CLI [az ad sp create-for-rbac](/cli/azure/ad/sp?view=azure-cli-latest#az-ad-sp-create-for-rbac) command:
```

Also, why am I creating a service principal here? If I'm following along, I already have one with all data plane permissions for my vault.